docs/design/generics/overview.md

```diff
@@ -43,51 +43,79 @@ pointers to other design documents that dive deeper into individual topics.
     signatures.
 -   Clearer and earlier error messages.
 -   Fast builds, particularly development builds.
 -   Support for both static and dynamic dispatch.
 
+Checked generics do have some restrictions, but are expected to be more
+appropriate at public API boundaries than templates.
+
 For more detail, see [the detailed discussion of generics goals](goals.md) and
 [generics terminology](terminology.md).
 
 ## Summary
 
 Summary of how Carbon generics work:
 
--   _Generics_ are parameterized functions and types that can apply generally.
-    They are used to avoid writing specialized, near-duplicate code for similar
-    situations.
--   Generics are written using _interfaces_ which have a name and describe
-    methods, functions, and other entities for types to implement.
+-   _Generics_ are compile-time parameterized functions, types, and other
+    language constructs. Those parameters allow a single definition to apply
+    more generally. They are used to avoid writing specialized, near-duplicate
+    code for similar situations.
+-   The definition of a _checked_ generic is typechecked once, without having to
+    know the specific argument values of the generic parameters it is
+    instantiated with. Typechecking the definition of a checked generic requires
+    a precise contract specifying the requirements on the argument values.
+-   For parameters that will be used as types, those requirements are written
+    using _interfaces_. Interfaces have a name and describe methods, functions,
+    and other entities for types to implement.
 -   Types must explicitly _implement_ interfaces to indicate that they support
-    its functionality. A given type may implement an interface at most once.
--   Implementations may be part of the type's definition, in which case you can
-    directly call the interface's methods on those types. Or, they may be
-    external, in which case the implementation is allowed to be defined in the
-    library defining the interface.
--   Interfaces are used as the type of a generic type parameter, acting as a
-    _type-of-type_. Type-of-types in general specify the capabilities and
-    requirements of the type. Types define specific implementations of those
-    capabilities. Inside such a generic function, the API of the type is
-    [erased](terminology.md#type-erasure), except for the names defined in the
-    type-of-type.
--   _Deduced parameters_ are parameters whose values are determined by the
-    values and (most commonly) the types of the explicit arguments. Generic type
-    parameters are typically deduced.
--   A function with a generic type parameter can have the same function body as
-    an unparameterized one. Functions can freely mix generic, template, and
-    regular parameters.
+    their functionality. A given type may implement an interface at most once.
+-   Implementations may be declared inline in the body of a class definition, or
```

Comment:
I read this as an exclusive "or", which doesn't seem right, given that the same implementation can be declared in both places. Maybe removing the comma would help:
```suggestion
-   Implementations may be declared inline in the body of a class definition or
```